### ISSUE TITLE
Modal prompts on iOS overlay the toast

### ISSUE THREAD

**Issue by Kief5555** [calintamas/react-native-toast-message#494]
**Describe the bug**
When a modal is prompted on ios calling `Toast.show` makes the toast appear UNDER the modal resulting in a 50% view of the toast (because of the modal that is blocking it)

**Steps to reproduce**
```<ThemeProvider value={colorScheme ? DarkTheme : DefaultTheme}>
				<Stack>
					<Stack.Screen name="(tabs)" options={{headerShown: false, headerBackTitle: ''}} />
					<Stack.Screen
						name="example"
						options={{
							title: example,
							presentation: 'modal',
						}}
				          />
			                </Stack>
			                </ThemeProvider>
			<Toast config={toastConfig} />
					```
Open the modal on iOS call toast.show
**Expected behavior**
For it to overlay/go above the modal
**Screenshots**
![image](https://github.com/calintamas/react-native-toast-message/assets/67562560/c8bab246-ae6f-453e-a949-5ecf92256911)
Above the moda, I have a custom toast configuration.
**Code sample**
Using react-nativgation and expo router (default tabs template with expo)

**Environment (please complete the following information):**
 - OS: iOS
 - react-native-toast-message version: latest
 - react-native version SDK 49

**Comment by hackerhgl** [comment]
@Kief5555 Were you able to figure out a solution?

**Comment by Kief5555** [comment]
> @Kief5555 Were you able to figure out a solution?

No, I might try to import the toast in the modal and try to display it there.

**Comment by jmcartlamy** [comment]
Are you tried this ? https://github.com/calintamas/react-native-toast-message/blob/main/docs/modal-usage.md#notes-regarding-react-native-modal-or-nativestacknavigator

**Comment by devoren** [comment]
@jmcartlamy Hey! I use expo router so if i use like this i have a warning. Also i see this comment https://github.com/calintamas/react-native-toast-message/issues/195#issuecomment-890760286 so i will try **containedModal**

**Comment by QuixThe2nd** [comment]
Im getting the same issue too.

**Comment by hichemfantar** [comment]
On IOS you have declare a new toast element inside each modal, refs are auto handled
https://github.com/calintamas/react-native-toast-message/blob/main/docs/modal-usage.md#notes-regarding-react-native-modal-or-nativestacknavigator

**Comment by Kief5555** [comment]
> On IOS you have create a new toast element inside each modal, refs are auto handled
> 
> https://github.com/calintamas/react-native-toast-message/blob/main/docs/modal-usage.md#notes-regarding-react-native-modal-or-nativestacknavigator

This is a different type of modal, I think. Shown in the screenshot is the native Modal Component (not view)

**Comment by hichemfantar** [comment]
> > On IOS you have create a new toast element inside each modal, refs are auto handled
> > https://github.com/calintamas/react-native-toast-message/blob/main/docs/modal-usage.md#notes-regarding-react-native-modal-or-nativestacknavigator
> 
> This is a different type of modal, I think. Shown in the screenshot is the native Modal Component (not view)

It's the same for both stack modals and the RN Modal component.

**Comment by moritzlang** [comment]
Rendering content on top of native views (`presentation="modal"`) is a bit tricky. What works for me is rendering the `Toast` component again in the Screen using `@gorhom/portal` as following:
```jsx
<Portal hostName="toast">
  <FullWindowOverlay>
    <Toast />
  </FullWindowOverlay>
</Portal>
```
Be aware that this solution does not work when running the app on the simulator. It somehow only works on the final build.
See [React Native Screens Integration](https://github.com/gorhom/react-native-portal?tab=readme-ov-file#react-native-screens-integration) docs for more details.

**Comment by calummoore** [comment]
For me, it worked when I added the Toast to the modal view, but the Toast has to be the last component.

```tsx
export function ModalView({ children }: ModalViewProps) {
  return (
    <>
      {children}
      <Toast />
    </>
  )
}

```

**Comment by alexandrius** [comment]
Use `FullWindowOverlay` from `react-native-screens`

**Comment by lorenzomilesi** [comment]
The usage of `FullWindowOverlay` breaks accessibility

**Comment by robert-barahona** [comment]
The same is happening for our team 😕 we tried to put `<Toast />` inside my screen but didn't work.

**Comment by edward99enclave** [comment]
> For me, it worked when I added the Toast to the modal view, but the Toast has to be the last component.
> 
> export function ModalView({ children }: ModalViewProps) {
>   return (
>     <>
>       {children}
>       <Toast />
>     </>
>   )
> }

It works for me.
